samtranslator/model/api/api_generator.py

```diff
@@ -626,14 +665,20 @@ def _construct_alias_target(self, domain: Dict[str, Any], api_domain_name: str,
         return alias_target
 
     @cw_timer(prefix="Generator", name="Api")
-    def to_cloudformation(self, redeploy_restapi_parameters, route53_record_set_groups):  # type: ignore[no-untyped-def]
+    def to_cloudformation(
+        self, redeploy_restapi_parameters: Optional[Any], route53_record_set_groups: Dict[str, Route53RecordSetGroup]
+    ) -> List[Any]:
```

Comment:
Any better types?

Reply:
That reminds me we can move un-nest step into the `to_cloudformation` so we can use `List<Resource>` here. Updated.